Library/Homebrew/formula.rb

```diff
@@ -775,6 +775,8 @@ def to_hash
           "used_options" => tab.used_options.as_flags,
           "built_as_bottle" => tab.built_bottle,
           "poured_from_bottle" => tab.poured_from_bottle
+          if ENV["HOMEBREW_CXX"]: "CXX_Compiler" => ENV["HOMEBREW_CXX"]
```

Comment:
I'm using boost and boost-python for a project interfacing python with C libraries which have lots of old C/C++ code involved, and I've found that to build the  I need to install boost with HOMEBREW_CXX as g++-4.9.  Other components of this project have compiler flags which only work with clang, and I often must switch to clang boost. Its hard to keep track of which boost version I currently have installed, and being able to access this info without digging through the Homebrew logs seems useful (honestly I did not think to look in `~/Library/Logs/Homebrew/boost`... Looking in there now, its not immediately clear how I can access the `$HOMEBREW_CXX` flag used).